src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/EcmaFormat/ReflectionTypeProvider.cs

```diff
@@ -0,0 +1,306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
```

Comment:
These aren't really helpers intended for general use. This is an object that implements an interface which is used to specialize a generic bit of code. I've rewritten the names of the methods to make them explicit interface implementations which should make that a bit more clear.